### ISSUE TITLE
[CT-2692] [Bug]  cannot import name 'DEFAULT_PROFILES_DIR' from 'dbt.config.profile'

### ISSUE THREAD

**Issue by danielefrigo** [dbt-labs/dbt-core#7855]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I tried upgrading my dbt project from 1.4 to 1.5.
Once installed the new version using pip, dbt doesn't work anymore, raising this error message:

`Traceback (most recent call last):
  File "/usr/local/bin/dbt", line 5, in <module>
    from dbt.main import main
  File "/usr/local/lib/python3.8/site-packages/dbt/main.py", line 45, in <module>
    from dbt.config.profile import DEFAULT_PROFILES_DIR, read_user_config
ImportError: cannot import name 'DEFAULT_PROFILES_DIR' from 'dbt.config.profile' (/home/vscode/.local/lib/python3.8/site-packages/dbt/config/profile.py)`

I tried also completely removing dbt 1.4 and installing 1.5 from scratch, but with the same result.

### Expected Behavior

should work like with 1.4 version

### Steps To Reproduce

1. install dbt 1.5
2. run `dbt debug`

### Environment

```markdown
- OS: Linux (devcontainer based on mcr.microsoft.com/vscode/devcontainers/python:0-3.8-bullseye)
- Python: 3.8.16
- dbt: 1.5.1 (tried also with 1.5.0)
```

### Which database adapter are you using with dbt?

snowflake

**Comment by dbeatty10** [comment]
Thanks for reaching out @danielefrigo!

I think the last version of dbt that included [that line of code](https://github.com/dbt-labs/dbt-core/blob/1.2.latest/core/dbt/main.py#L45) was dbt-core v1.2.

So I think there's some old version of dbt code hanging around in your system somehow (rather than this being a bug in dbt-core).

If you are using pip with a shell like `zsh` or `bash`, could you try the following?
```shell
python -m venv .env
source .env/bin/activate
python -m pip install --upgrade pip
python -m pip install --upgrade dbt-snowflake
source .env/bin/activate
dbt --version
dbt debug
deactivate
```

You can delete that virtual environment like this when you are done with it:
```shell
rm -rf .env
```

**Comment by danielefrigo** [comment]
You're absolutely right, previously we were using dbt 1.2.
I tried completely recreating the venv and the issue has gone away!
Thanks

**Comment by katepaydhealth** [comment]
I got the same issue with @danielefrigo. I have tried to create the venv and follow the steps @dbeatty10 showed above then the issue went away. Unfortunately, when I deactivated the venv, I got the issue again. Please advices!
dbt version: 1.5.2
dbt-snwoflake: 1.5.2

**Comment by dbeatty10** [comment]
@katepaydhealth if the venv instructions worked for you, then running just the following should get your system fully up-to-date with 1.6.0:

```shell
python -m pip install --upgrade pip
python -m pip install --upgrade dbt-snowflake
```

If there's a lower version you want to use instead (like 1.4), then you can do [something like this](https://docs.getdbt.com/docs/core/pip-install):
```shell
python -m pip install --upgrade pip
python -m pip install --upgrade "dbt-snowflake~=1.4.0"
```

**Comment by dwmintz** [comment]
Just to keep things interesting, I tried upgrading dbt-core and dbt-bigquery from 1.1 to latest and ran into this error. The fresh venv trick sadly did not work. But downgrading to 1.4 (or 1.3 or 1.2) fixed me. Any upgrade that takes me to 1.5+ causes this error and I haven't figured out how to get past it.

**Comment by talarczykco** [comment]
**If using VS Code**, make sure the version of the extension is recent enough to match the version of dbt. I recently encountered a cloud development environment that was using an old version of `code-server` that limited my marketplace extensions; no error was given from the marketplace, just old versions of the extension. Once I updated code-server, I was able to install a recent enough extension that it fixed this problem.
